src/index.js

```diff
@@ -1,22 +1,112 @@
 import { registerBlockType } from '@wordpress/blocks';
-import { useBlockProps } from '@wordpress/block-editor';
+import {
+        useBlockProps,
+        InspectorControls,
+} from '@wordpress/block-editor';
+import {
+        PanelBody, 
+        TextControl,
+} from '@wordpress/components';
+import apiFetch from '@wordpress/api-fetch';
 
-const blockStyle = {
-    backgroundColor: '#008710',
-    color: '#fff',
-    padding: '20px',
-};
+registerBlockType( 'kokkieh/gh-commit-block', {
+    icon: <svg><path d="M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12"/></svg>,
+    attributes: {
+        commitHash: {
+            type: 'string',
+        },
+        commitSha: {
+            type: 'string',
+        },
+        commitUrl: {
+            type: 'string',
+        },
+        authorName: {
+            type: 'string',
+        },
+        authorUrl: {
+            type: 'string',
+        },
+        commitMessage: {
+            type: 'string',
+        },
+    },
+
+    edit: ( { attributes, setAttributes } ) => {
+        const blockProps = useBlockProps( { className: 'blockStyles' } );
+        const {
+            commitHash,
+            commitSha,
+            commitUrl,
+            authorName,
+            authorUrl,
+            commitMessage,
+        } = attributes;
+
+        // Update commitHash value when entered into block settings
+        function onChangeTextField( newValue ) {
+            setAttributes( { commitHash: newValue } );
+            renderCommit(newValue);
+        }
+
+        // Function to make API request and render the commit data
+        // (only called when the commit hash is updated in block settings)
+        function renderCommit(hash) {
+            apiFetch( { path: '/kokkieh/commit/' + hash } )
+            .then( ( data ) => {
+                setAttributes( { commitSha: data.sha } );
+                setAttributes( { commitUrl: data.html_url } );
+                setAttributes( { authorName: data.commit.author.name } );
+                setAttributes( { authorUrl: data.author.html_url } );
+                setAttributes( { commitMessage: data.commit.message } );
+            } )
+            .catch( error => {
+                setAttributes( { commitSha: "Please enter a valid commit ID." } );
```

Comment:
Wrapping the text constant with a translation function would make it even better.

Reply:
I plan to add translation in a different PR: I need to go through the entire plugin and make sure all strings are internationalized and using the correct text-domain, and at this point it makes more sense to me to do that all at once to ensure consistency.